client.go

```diff
@@ -275,3 +288,65 @@ func (client *Client) Info(key string) (res KeyInfo, err error) {
 	res, err = ParseInfo(conn.Do("TS.INFO", key))
 	return res, err
 }
+
+// Get all the keys matching the filter list.
+func (client *Client) QueryIndex(filters ...string) (keys []string, err error) {
+	conn := client.Pool.Get()
+	defer conn.Close()
+
+	if len(filters) == 0 {
+		return
+	}
+
+	args := redis.Args{}
+	for _, filter := range filters {
+		args = args.Add(filter)
+	}
+	return redis.Strings(conn.Do("TS.QUERYINDEX", args...))
+}
+
+// Creates a new sample that increments the latest sample's value
+func (client *Client) IncrBy(key string, timestamp int64, value float64, options CreateOptions) (int64, error) {
+	conn := client.Pool.Get()
+	defer conn.Close()
+
+	args, err := AddCounterArgs(key, timestamp, value, options)
+	if err != nil {
+		return -1, err
+	}
+	return redis.Int64(conn.Do("TS.INCRBY", args...))
+}
+
+// Creates a new sample that decrements the latest sample's value
+func (client *Client) DecrBy(key string, timestamp int64, value float64, options CreateOptions) (int64, error) {
+	conn := client.Pool.Get()
+	defer conn.Close()
+
+	args, err := AddCounterArgs(key, timestamp, value, options)
+	if err != nil {
+		return -1, err
+	}
+	return redis.Int64(conn.Do("TS.DECRBY", args...))
+}
+
+// Add counter args for command TS.INCRBY/TS.DECRBY
+func AddCounterArgs(key string, timestamp int64, value float64, options CreateOptions) (redis.Args, error) {
+	args := redis.Args{key, value}
+	if timestamp > 0 {
+		args = args.Add("TIMESTAMP", timestamp)
+	}
+
+	return options.Serialize(args)
+}
+
+// Append new samples to a list of series.
+func (client *Client) MultiAdd(samples ...Sample) ([]interface{}, error) {
```

Comment:
hi there @dengliming the TS.MADD accepts multiple keys as input. Your approach is correct, but does not allow for samples from different series.
example:
```
127.0.0.1:6379>TS.MADD temperature:2:32 1548149180000 26 cpu:2:32 1548149183000 54
```
happy to approve the other changes while we work on MADD in another PR, or to fix this one :) you decide what suits you best.